Wip0.9.0

bidoubiwa opened the pull request.

curquiza commented:
I don't see any "add_or_update_documents" method (PUT). This method should give the possibility to add a primary-key.

curquiza commented on src/__tests__/meili.ts:
```diff
@@ -73,56 +79,45 @@ test('reset-start', async () => {
     uid: 'abababa',
   }
+  const indexIndentifier = {
+    uid: 'bebebebe',
```
Could we please use more "professional" example? 😇 Like `index_uid` or just `uid`?

bidoubiwa replied:
Done

curquiza commented:
Is the option `apiKey` still used?
Because it should not, the SDK should use `masterKey` instead.
I saw the mistake in the README, in:
- the docker line
- the getting started example (I succeed to add a suggestion)

curquiza commented:
For the methods `getUpdate` and `getUpdates`:
For me, they should be renamed as `getUpdateStatus` and `getAllUpdateStatus` to be consitent with the documentation titles.

qdequele commented:
There is an issue with the test. Could not connect to the docker?

bidoubiwa commented:
@qdequele It was because I added the --master-key to the variables passed to MeiliSearch. But because docker downloads v0.8.4  the master key was still called api-key, which results in an error and thus the server does not start.

I put it there in preparation of the 0.9.0, I removed it so we can see which test fails with the old version.

bidoubiwa commented on README.md:
```diff
@@ -38,205 +49,244 @@ NB: you can also download MeiliSearch from **Homebrew** or **APT**.
-})
+### Indexes
+
+#### Create an index <!-- omit in toc -->
```
It's a plugin in Visual Studio Code, that lets you generate a Table of Contents. By adding this comment it does not add that title.

qdequele replied:
Normally it's not a good idea to add on a public code, specific IDE things.

curquiza replied:
It's not specific to an IDE.
I use it with my VScode plugin, but you can use what you want. I saw some opensources projects use this package for example: https://github.com/thlorenz/doctoc

bidoubiwa commented on src/indexes.ts:
```diff
@@ -210,14 +198,35 @@ class Indexes {
   /**
-   * Add or update multiples documents to an index
+   * Add or replace multiples documents to an index
    * @memberof Documents
```
Should I remove all `@memberof`

qdequele replied:
No, but it's best to put the good "member of".

qdequele replied:
This one is probably a member of Indexes Class?

bidoubiwa commented on .github/workflows/test.yml:
```diff
@@ -9,7 +9,7 @@ jobs:
       - uses: actions/checkout@v1
       - name: Docker setup
-        run: docker run -d -p 7700:7700 getmeili/meilisearch:v0.8.4 ./meilisearch --no-analytics=true
+        run: docker run -d -p 7700:7700 getmeili/meilisearch:v0.8.4 ./meilisearch --no-analytics=true --master-key=123
```
Clementine asked for consistency through all SDK's and apparently she tries with the master key.

qdequele commented on src/types.ts:
```diff
@@ -29,34 +15,39 @@ export interface AsyncUpdateId {
+  rankingDistinct?: string
+  searchableAttributes?: string[]
+  displayedAttributes?: string[]
   distinctField?: string
```
```suggestion
  distinctAttribute?: string
  searchableAttributes?: string[]
  displayedAttributes?: string[]
```

curquiza commented:
About the tests:
- There is no test when it does not work, for example when it should raise an exception 
- there is still comments with "todo" things

curquiza commented:
Why the "Index" method does start with a "i" in uppercase?
For me, the convention in JS is to be in lower camelCase

bidoubiwa commented on src/indexes.ts:
```diff
@@ -255,52 +269,282 @@ class Indexes {
   }
 
+  /**
+   * Update all settings. Any parameters not provided will be left unchanged.
```
Why not? Should it be on two lines? Is the explanation too long?

bidoubiwa commented on src/indexes.ts:
```diff
@@ -232,16 +241,21 @@ class Indexes {
    * @memberof Documents
    * @method deleteDocuments
    */
   deleteDocuments(documentsIds: string[]): Promise<Types.AsyncUpdateId> {
```
This is not correct, I tried it. It is : ` documentsIds: string[] | number[]`

curquiza commented on README.md:
```diff
@@ -38,212 +49,248 @@ NB: you can also download MeiliSearch from **Homebrew** or **APT**.
-  "updateId": 1
+  "hits": [
+    {
+      "title": "Le Petit Prince",
```
```suggestion
      "book_id": 456,
      "title": "Le Petit Prince",
```

curquiza commented on README.md:
```diff
@@ -38,212 +49,248 @@ NB: you can also download MeiliSearch from **Homebrew** or **APT**.
+    {
+      "title": "Le Petit Prince",
+      "_formatted": {
+        "title": "Le Petit <em>Prince</em>"
```
```suggestion
        "book_id": 456,
        "title": "Le Petit <em>Prince</em>"
```
